nautobot/ipam/models.py

```diff
@@ -899,6 +919,21 @@ class IPAddress(PrimaryModel, StatusModel, RoleModelMixin):
     )
     broadcast = VarbinaryIPField(null=False, db_index=True, help_text="IPv4 or IPv6 broadcast address")
     prefix_length = models.IntegerField(null=False, db_index=True, help_text="Length of the Network prefix, in bits.")
+    parent = models.ForeignKey(
+        "ipam.Prefix",
+        blank=True,
+        null=True,
+        related_name="ip_addresses",  # `IPAddress` to use `related_name="ip_addresses"`
+        on_delete=models.PROTECT,
+        help_text="The parent Prefix of this IPAddress.",
+    )
+    # ip_version is set internally just like network, broadcast, and prefix_length.
+    ip_version = models.IntegerField(
+        choices=choices.IPAddressVersionChoices,
+        null=True,
```

Comment:
why null=true?

Reply:
Same "it wants a default value if it's a choice field unless it's nullable" problem as always...

It's not user-serviceable and always forcefully set so I'm not worried about this one so much.